src/clj/clojure/core.clj

```diff
@@ -4037,7 +4037,7 @@
    :static     true
    :deprecated "1.9"}
   [ns]
-  (.getName (the-ns ns)))
+  (symbol (.getName (the-ns ns))))
```

Comment:
I did wonder about changing the return type, but thought there was a good reason for it.